tough/src/editor.rs

```diff
@@ -0,0 +1,660 @@
+                }
+            }
+            RoleType::Timestamp => "timestamp.json".to_string(),
+            RoleType::Root => format!("{}.root.json", self.signed.signed.version()),
+        };
+
+        let path = outdir.join(filename);
+        std::fs::write(&path, &self.buffer).context(error::FileWrite { path })
+    }
+}
+
+/// A set of signed TUF Repository metadata.
+///
+/// This metadata represents a signed TUF repository, but without the actual
+/// targets, the repository cannot be used. It is up to ther user to ensure
+/// all the target files are in the proper directories along with the
+/// contained metadata. There are convenience methods provided to help with
+/// this.
+#[derive(Debug)]
+pub struct SignedRepository {
+    root: SignedRole<Root>,
+    targets: SignedRole<Targets>,
+    snapshot: SignedRole<Snapshot>,
+    timestamp: SignedRole<Timestamp>,
+}
+
+impl SignedRepository {
+    /// Writes the metadata to the given directory. If consistent snapshots
+    /// are used, the appropriate files are prefixed with their `sha256`.
+    pub fn write<P>(&self, outdir: P) -> Result<()>
+    where
+        P: AsRef<Path>,
+    {
+        let consistent_snapshot = self.root.signed.signed.consistent_snapshot;
+        self.root.write(&outdir, consistent_snapshot)?;
+        self.targets.write(&outdir, consistent_snapshot)?;
+        self.snapshot.write(&outdir, consistent_snapshot)?;
+        self.timestamp.write(&outdir, consistent_snapshot)
+    }
+
+    /// Crawls a given directory and symlinks any targets found to the given
+    /// "out" directory.
+    ///
+    /// For each file found in the `indir`, the method gets the filename and
+    /// if the filename exists in `Targets`, the file's sha256 is compared
+    /// against the data in `Targets`. If this data does not match, the
+    /// method will fail. If all is well, the taret is symlinked.
+    pub fn link_targets<P1, P2>(&self, indir: P1, outdir: P2) -> Result<()>
+    where
+        P1: AsRef<Path>,
+        P2: AsRef<Path>,
+    {
+        std::fs::create_dir_all(&outdir).context(error::DirCreate {
+            path: &outdir.as_ref(),
+        })?;
+        // Get the absolute path of the indir and outdir
+        let abs_indir = std::fs::canonicalize(&indir).context(error::AbsolutePath {
+            path: indir.as_ref().to_owned(),
+        })?;
+        let abs_outdir = std::fs::canonicalize(&outdir).context(error::AbsolutePath {
+            path: outdir.as_ref().to_owned(),
+        })?;
+        let repo_targets = &self.targets.signed.signed.targets;
+
+        // Walk the absolute path of the indir. Using the absolute path here
+        // means that `entry.path()` call will return its absolute path.
+        let walker = WalkDir::new(abs_indir).follow_links(false);
+        for entry in walker {
+            let entry = entry.context(error::WalkDir)?;
+
+            // If the entry is a file...
+            if entry.file_type().is_file() {
```

Comment:
```suggestion
            if !entry.file_type().is_file() {
                continue;
            }
```
Rather than the `else` way below.